iothub_client/inc/iothub_twin.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/** @file   iothub_twin.h
+*   @brief  The @c IoTHub_Twin component encapsulates all twin message related information that can
+*           be transferred or received by an IoT hub client.
+*/
+
+#ifndef IOTHUB_TWIN_H
+#define IOTHUB_TWIN_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+extern "C"
+{
+#endif
+
+
+#ifndef IOTHUB_TWIN_REQUEST_OPTIONS_TYPE
+typedef struct IOTHUB_TWIN_REQUEST_OPTIONS_TAG* IOTHUB_TWIN_REQUEST_OPTIONS_HANDLE;
+#define IOTHUB_TWIN_REQUEST_OPTIONS_TYPE
+#endif
+#ifndef IOTHUB_TWIN_RESPONSE_TYPE
+typedef struct IOTHUB_TWIN_RESPONSE_TAG* IOTHUB_TWIN_RESPONSE_HANDLE;
+#define IOTHUB_TWIN_RESPONSE_TYPE
+#endif
+
+/**
+ * @brief  GET twin request options
+ */
+typedef bool(*IOTHUB_TWIN_REQUEST_OPTIONS_GET_INT64)(IOTHUB_TWIN_REQUEST_OPTIONS_HANDLE twin_request_options, int64_t* value);
+typedef void(*IOTHUB_TWIN_REQUEST_OPTIONS_SET_INT64)(IOTHUB_TWIN_REQUEST_OPTIONS_HANDLE twin_request_options, int64_t* value);
+
+typedef struct IOTHUB_TWIN_REQUEST_OPTIONS_TAG
+{
+    /**
+     * @brief Get current_version for @c IOTHUB_TWIN_REQUEST_OPTIONS
+     *
+     * @param twin_request_options  Pointer to self.
+     * @param value                 Value of current_version.
+     * @return true if current_version has been set.
+     * @return false if current_version has not been set. This means the option not used.
+     */
+    IOTHUB_TWIN_REQUEST_OPTIONS_GET_INT64 get_current_version;
+
+    /**
+     * @brief Set current_version for @c IOTHUB_TWIN_REQUEST_OPTIONS
+     *
+     * @param twin_request_options  Pointer to self.
+     * @param value                 Pointer to value to set the current_version.
+     *                              May be NULL if option to not be used.
```

Comment:
May be NULL if option to not be used?

Reply:
I'll rephrase.  If the user wants to turn the option "off", they can pass in NULL, instead of using a pointer to an lvalue ... which turns the option "on" and sets the option value to the lvalue.